### DISCUSSION TITLE
This is a proxy for the unknown type 'Bonsai.Scripting.ExpressionTransform'. Please install any missing packages or replace this operator.

### DISCUSSION THREAD

**Discussion by SamuelLiebana** [bonsai-rx/bonsai#1558] [Q&A] [answered]
We have been experiencing the following error on updating from Bonsai 2.6.3 to 2.8:

"This is a proxy for the unknown type 'Bonsai.Scripting.ExpressionTransform'. Please install any missing packages or replace this operator"

This is when we try to run a Neurophotometrics script using the packages that they recommend downloading (Bonsai Starter Pack and Neurophotometrics.Design).

**Comment by SamuelLiebana** [comment, marked as answer]
Just managed to solve the issue by installing the Bonsai Scripting Library package. We recommend that either the starter pack is changed or Neurophotometric's manual is updated to reflect this extra step in the set-up process to get Neurophotometrics rigs working with the new version of Bonsai.

**Reply by bruno-f-cruz** [reply]
The `Bonsai.Scripting` package has split in two. `Bonsai.Scripting.Expressions` (contains the `ExpressionTransform` operator) and the `Bonsai.Scripting.IronPython`, which contains the IronPython package. This decision was mainly due to the existence of an alternative Python package (Bonsai.Scripting.Python), and the fact that several users rarely use the IronPython expressions to begin with. As this package is fairly large compared to a normal bonsai package (100x larger), a decision was made to separate them and let users decide which package they want to install. 

Not sure who is maintaining this Neurophotometrics package, but if this is an issue I would urge you to get in touch with them and let them know about this change.
